frontend/src/components/Modals/MangeWorkspace/Documents/index.jsx

```diff
@@ -146,6 +148,21 @@ export default function DocumentSettings({
       !systemSettings?.EmbeddingEngine ||
       systemSettings.EmbeddingEngine === "openai"
     ) {
+      const embeddingModel =
+        process.env.EMBEDDING_MODEL_PREF || "text-embedding-ada-002";
+      let COST_PER_TOKEN;
+
+      switch (embeddingModel) {
+        case "text-embedding-3-small":
+          COST_PER_TOKEN = TEXT_EMBEDDING_3_SMALL_COST_PER_TOKEN;
+          break;
+        case "text-embedding-3-large":
+          COST_PER_TOKEN = TEXT_EMBEDDING_3_LARGE_COST_PER_TOKEN;
+          break;
+        default:
+          COST_PER_TOKEN = ADA_COST_PER_TOKEN;
+          break;
+      }
```

Comment:
@shatfield4 ^ Do what is suggested by the bot here. Its much more readable and maintainable to run a map since its simple Key:value comparison

My only change is add the fallback as you had for the default case

```
const COST_PER_TOKEN = MODEL_COSTS[embeddingModel || "text-embedding-ada-002"];
```

Reply:
Actually, i see `embeddingModel` already does the fallback. You could drop that variable totally if you wanted. Whatever looks best

Reply:
Fixed and cleaned this up even more so it's all in one model object 👍